scene/resources/texture.cpp

```diff
@@ -823,6 +823,8 @@ void ImageTexture3D::_update(const TypedArray<Image> &p_data) {
 }
 
 Error ImageTexture3D::create(Image::Format p_format, int p_width, int p_height, int p_depth, bool p_mipmaps, const Vector<Ref<Image>> &p_data) {
+	Image::Image3DValidateError verr = Image::validate_3d_image(p_format, p_width, p_height, p_depth, p_mipmaps, p_data);
+	ERR_FAIL_COND_V(verr != Image::VALIDATE_3D_OK, ERR_CANT_CREATE);
```

Comment:
`validate_3d_image` may be a costly operation, I'd wrap this up to be running in debug builds only:

```suggestion
#ifdef DEBUG_ENABLED
	Image::Image3DValidateError verr = Image::validate_3d_image(p_format, p_width, p_height, p_depth, p_mipmaps, p_data);
	ERR_FAIL_COND_V(verr != Image::VALIDATE_3D_OK, ERR_CANT_CREATE);
#endif
```

Reply:
That means that this PR doesn't fix the crash in release builds though.

Reply:
A game project should be developed without introducing run-time errors, especially for core stuff like this, otherwise we end up with slow release builds...

Reply:
this check is actually done twice now, we could remove the one in `RendererStorageRD::texture_3d_initialize` if that doesn't any problems

Reply:
I'm not sure why this is needed in the first place, shouldn't `RenderingServer::get_singleton()->texture_3d_create()` return an invalid `RID` if the parameters are wrong, and thus trigger the early return below?

But also, is `texture_3d_create` actually implemented? The only reference I find is:
```
servers/rendering_server.h:     virtual RID texture_3d_create(Image::Format, int p_width, int p_height, int p_depth, bool p_mipmaps, const Vector<Ref<Image>> &p_data) = 0; //all slices, then all the mipmaps, must be coherent
```
But there's no actual implementation.

Reply:
It should be defined in
```FUNCRIDTEX6(texture_3d, Image::Format, int, int, int, bool, const Vector<Ref<Image>> &)```
right?